### DISCUSSION TITLE
TypeError after upgrading from v84.4.10 to v86.0.241

### DISCUSSION THREAD

**Discussion by veronikavychodilova** [cefsharp/CefSharp#3375] [Q&A] [answered]
Hi, I've tried to upgrade cefSharp.Winforms from v84.4.10 to latest stable (v86.0.241) and after that the binding stopped working - I was getting following error: TypeError: Cannot read property 'EditorInit' of undefined.

I've only changed the original code:
```
WebBrowser.JavascriptObjectRepository.Register("editorInit", _editorInit, true, new BindingOptions { CamelCaseJavascriptNames = false }); 
WebBrowser.JavascriptObjectRepository.Register("codeChange", _codeChange, true, new BindingOptions { CamelCaseJavascriptNames = false }); 
WebBrowser.JavascriptObjectRepository.Register("editorOpen", _editorOpen, true, new BindingOptions { CamelCaseJavascriptNames = false });
```
to new code (as stated in suggestion under ticket #3141):
```
WebBrowser.JavascriptObjectRepository.NameConverter = null; WebBrowser.JavascriptObjectRepository.Register("editorInit", _editorInit, true, new BindingOptions { }); WebBrowser.JavascriptObjectRepository.Register("codeChange", _codeChange, true, new BindingOptions { }); 
WebBrowser.JavascriptObjectRepository.Register("editorOpen", _editorOpen, true, new BindingOptions { });
```

Am I missing something there? Thank you.

**Comment by amaitland** [comment, marked as answer]
> I was getting following error: TypeError: Cannot read property 'EditorInit' of undefined.

What does your javascript look like?

Based on this error it looks like you have a case mismatch, you are registering `editorInit` and attempting to access `EditorInit` (noting the difference in capitalisation for the first letter). 

The following would match the naming of based on the error message.
```c#
WebBrowser.JavascriptObjectRepository.Register("EditorInit", _editorInit, true, new BindingOptions { });
```

Not clear this has anything to do with the changes in #3141 as the method and property names should have been the only to have been affected.

**Reply by veronikavychodilova** [reply]
The script was the same all the time (when using v84... and v86...) and it was working in v84.
It seems `WebBrowser.JavascriptObjectRepository.Register("EditorInit", _editorInit, true, new BindingOptions { });` does not work either...
My colleague who is/was developing the script part suggested to add debug log right before the line on which the type error is raised, so I've added: 
`Object.getOwnPropertyNames(window).forEach(function(v, x) {console.log(v);});
window["editorInit"].EditorInit("true"); \\TypeError is raised on that line`
Neither "editorInit" nor "EditorInit" was present, so the registration wasn't performed at all.

**Reply by amaitland** [reply]
What does your JavaScript look like? Where are you calling CefSharp.BindObjectAsync?

**Reply by veronikavychodilova** [reply]
I'm calling it from constructor of my control where the ChromiumWebBrowser is placed:
```
public EditorControlNew(IAlertServiceAdapter alertServiceAdapter, IPortProvider portProvider)
{
    _port = portProvider.GetPort();
    _logger.Debug($"new editor control instance, port provided={_port}");
    InitializeComponent();\\WebBrowser instance is created from there as this.WebBrowser = new ChromiumWebBrowser($"http://localhost:{_port}");

     _alertServiceAdapter = alertServiceAdapter;

     _codeChange = new CodeChange();
     _codeChange.Changed += CodeChange_Changed;

      _editorInit = new EditorInitialization();
      _editorInit.Initialized += EditorInit_Initialized;

       _editorOpen = new EditorOpening();
       _editorOpen.EditorOpened += EditorOpen_EditorOpened;

        CefSharpSettings.LegacyJavascriptBindingEnabled = true;
        CefSharpSettings.WcfEnabled = true;
        WebBrowser.JavascriptObjectRepository.NameConverter = null;
        WebBrowser.JavascriptObjectRepository.Register("editorInit", _editorInit, true, new BindingOptions { });
       WebBrowser.JavascriptObjectRepository.Register("codeChange", _codeChange, true, new BindingOptions { }); 
       WebBrowser.JavascriptObjectRepository.Register("editorOpen", _editorOpen, true, new BindingOptions { });

        WebBrowser.LoadError += WebBrowser_LoadError;
        WebBrowser.ConsoleMessage += WebBrowser_ConsoleMessage;
}
```
I've attached the script - the exception is happening on line 3004.
[main-es2015.zip](https://github.com/cefsharp/CefSharp/files/5896321/main-es2015.zip)

**Reply by amaitland** [reply]
You'll need to change the way you are configuring legacy JavaScript Binding. The easiest option is to use the following instead of CefSharpSettings.LegacyJavascriptBindingEnabled

```
chromiumWebBrowser.JavascriptObjectRepository.Settings.LegacyBindingEnabled
```

See https://github.com/cefsharp/CefSharp/issues/2977 for details.

**Reply by veronikavychodilova** [reply]
Thank you, Alex, for being helpful. It is working now.
